spec/features/admin/invoices/show_spec.rb

```diff
@@ -1,37 +1,63 @@
-      merchant_1 = create(:merchant)
-      customer_1 = create(:customer)
-      customer_2 = create(:customer)
-      invoice_1 = create(:invoice, customer: customer_1)
-      invoice_2 = create(:invoice, customer: customer_2)
+      visit admin_invoice_path(@invoice_1)
 
-      visit admin_invoice_path(invoice_1)
+      expect(page).to have_content("Invoice ID: #{@invoice_1.id}")
+      expect(page).to have_content("Invoice Status: #{@invoice_1.status}")
+      expect(page).to have_content("Invoice Created On: #{@invoice_1.created_at.strftime('%A, %B %d, %Y')}")
+      expect(page).to have_content("Customer Name: #{@customer_1.first_name} #{@customer_1.last_name}")
 
-      expect(page).to have_content("Invoice ID: #{invoice_1.id}")
-      expect(page).to have_content("Invoice Status: #{invoice_1.status}")
-      expect(page).to have_content("Invoice Created On: #{invoice_1.created_at.strftime('%A, %B %d, %Y')}")
-      expect(page).to have_content("Customer Name: #{customer_1.first_name} #{customer_1.last_name}")
-
-      expect(page).to_not have_content("Invoice ID: #{invoice_2.id}")
+      expect(page).to_not have_content("Invoice ID: #{@invoice_2.id}")
       expect(page).to_not have_content(
-        "Customer Name: #{invoice_2.customer.first_name} #{invoice_2.customer.last_name}"
+        "Customer Name: #{@invoice_2.customer.first_name} #{@invoice_2.customer.last_name}"
       )
 
-      visit admin_invoice_path(invoice_2)
+      visit admin_invoice_path(@invoice_2)
 
-      expect(page).to have_content("Invoice ID: #{invoice_2.id}")
-      expect(page).to have_content("Invoice Status: #{invoice_2.status}")
-      expect(page).to have_content("Invoice Created On: #{invoice_2.created_at.strftime('%A, %B %d, %Y')}")
-      expect(page).to have_content("Customer Name: #{customer_2.first_name} #{customer_2.last_name}")
+      expect(page).to have_content("Invoice ID: #{@invoice_2.id}")
+      expect(page).to have_content("Invoice Status: #{@invoice_2.status}")
+      expect(page).to have_content("Invoice Created On: #{@invoice_2.created_at.strftime('%A, %B %d, %Y')}")
+      expect(page).to have_content("Customer Name: #{@customer_2.first_name} #{@customer_2.last_name}")
 
-      expect(page).to_not have_content("Invoice ID: #{invoice_1.id}")
+      expect(page).to_not have_content("Invoice ID: #{@invoice_1.id}")
       expect(page).to_not have_content(
-        "Customer Name: #{invoice_1.customer.first_name} #{invoice_1.customer.last_name}"
-      )
+        "Customer Name: #{@invoice_1.customer.first_name} #{@invoice_1.customer.last_name}"
+)
+    end
+
+    it "displays all the Items on that invoice and all the items information" do
+      visit admin_invoice_path(@invoice_1)
+      
+      within("#item_#{@item_1.id}") do
+        expect(page).to have_content("Item name: #{@item_1.name}")
+        expect(page).to have_content("Item Sold Price: $#{@invoice_item_1.unit_price/ 100.to_f}")
```

Comment:
hey @AlexKiselich i think you may be missing a test for quantity ordered. I see it in the view but not in these within blocks. is it somewhere else?

Reply:
Great catch! Updated